docs/DOCS.md

```diff
@@ -0,0 +1,835 @@
+    - [4.1.5 Instructions](#415-instructions)
+    - [4.1.6 Notes and caveats](#416-notes-and-caveats)
+  - [4.2 Node.js (tests of JS code)](#42-nodejs-tests-of-js-code)
+- [5.0 Deploying](#50-deploying)
+  - [5.1 Development](#51-development)
+  - [5.2 Local development Docker infrastructure](#52-local-development-docker-infrastructure)
+    - [5.2.1 What](#521-what)
+    - [5.2.2 Why](#522-why)
+    - [5.2.3 How](#523-how)
+    - [5.2.4 Notes](#524-notes)
+    - [5.2.5 Troubleshooting](#525-troubleshooting)
+  - [5.2 Production](#52-production)
+    - [5.2.1 Gunicorn](#521-gunicorn)
+    - [5.2.2 Supervisord](#522-supervisord)
+- [6.0 Module `calendars`](#60-module-calendars)
+  - [6.1 Introductory notes](#61-introductory-notes)
+  - [6.2 Calendars](#62-calendars)
+  - [6.3 Datetimes (in specific calendars)](#63-datetimes-in-specific-calendars)
+  - [6.4 CF time systems](#64-cf-time-systems)
+  - [6.5 Usage](#65-usage)
+    - [6.5.1 `CalendarDatetime`: Datetimes in various calendars](#651-calendardatetime-datetimes-in-various-calendars)
+    - [6.5.2 `CfTimeSystem`: CF time systems](#652-cftimesystem-cf-time-systems)
+    - [6.5.3 `CfDatetime`: CF time points](#653-cfdatetime-cf-time-points)
+
+# 1.0 Dependencies
+
+## 1.1 Python
+
+The server for the PDP frontend is written in Python.
+It packages up, minifies, and serves the various "static" JS files and other resources
+(e.g., CSS) from which the frontend is built.
+
+The pdp requires that pip and tox be installed.
+
+```bash
+sudo apt-get install python-pip python-dev build-essential
+sudo pip install tox ## or pip install tox --user
+```
+
+Some of the required python libraries have system-level dependencies.
+
+```bash
+sudo apt-get install libhdf5-dev libnetcdf-dev libgdal-dev
+```
+
+And GDAL doesn't properly source it's own lib paths when installing the python package:
+
+```bash
+export CPLUS_INCLUDE_PATH=/usr/include/gdal
+export C_INCLUDE_PATH=/usr/include/gdal
+```
+
+## 1.2 Node.js (test framework only)
+
+The test framework for the PDP frontend runs in [Node.js](https://nodejs.org/en/).
+
+There are several different valid ways to install Node.js on a Ubuntu system.
+
+We reccomend using [nvm](https://github.com/creationix/nvm) to manage your node/npm install.
+It is a little more laborious (not a lot), and provides a lot more flexibility than the
+simpler installation methods, which you can look up by searching "ubuntu install nodejs".
+
+# 2.0 Installation
+
+## 2.1 Development
+
+### 2.1.1 Server
+
+With the prerequisites, creating a development environment should be as simple as:
+
+```bash
+git clone https://github.com/pacificclimate/pdp
+cd pdp
+tox -e devenv
+```
+
+It could take 5-30 minutes since tox will not use system packages and needs to build any package which requires it.
+
+### 2.1.2 JS tests
+
+With Node.js installed (see above), you can install all the test framework dependencies
+as follows:
+
+```bash
+npm install
+```
+
+Notes:
+
```

Comment:
Would these running-javascript-tests notes fit better in section 4.2 on running tests, rather than section 2.1 on installing things?